### PR TITLE
Add enums support

### LINKED ISSUE #86: Add enum support
Currently pulumi schema doesn't capture enum information.

### PR DESCRIPTION
**Pull request by mikhailshilkov**
Resolve #86 

Three questions that I needed to solve:

- Name of types. There's no name annotation for enums. If I use the property name as the type name, I get a bunch of clashes for common properties like `Type`. I went with a somewhat verbose `ParentTypeNamePropertyName` approach.
- Name of values. Values are usually UPPER_SNAKE_CASE or lowerCamelCase, so I decided to map them to our standard UpperCamelCase. See the list [here](https://gist.github.com/mikhailshilkov/060b4fe5cda202e248afefa6e3efb647), it makes sense to me.
- Strict vs. relaxed (allowing arbitrary values). I haven't found any guidance on that in DD documentation. I checked the Java and .NET SDKs and they _seem_ to use strict enums (or no enums at all in other places). I went ahead with strict enums. If we have to relax them, that's probably a less disruptive change than the other way around.

### CODE REVIEW

File: examples/pubsub-ts/index.ts [hunk 1]
```diff
@@ -48,7 +48,7 @@ const schema = new google.pubsub.v1.Schema("schema", {
         ],
     }),
     name: "my-schema",
-    type: "avro",
+    type: "AVRO",
```
Comment (lblackstone) [inline examples/pubsub-ts/index.ts:51]: Should this be an enum?

File: provider/pkg/gen/schema.go [hunk 2]
```diff
@@ -699,6 +701,56 @@ func (g *packageGenerator) genTypeSpec(typeName, propName string, prop *discover
...
+	// places of specs, so we want to error only if they a) have the same name
+	// b) the list of values does not match.
+	if other, ok := g.pkg.Types[tok]; ok {
+		same := len(enumSpec.Enum) == len(other.Enum)
+		for _, val := range other.Enum {
+			same = same && values.Has(val.Value.(string))
+		}
+		if !same {
+			return nil, errors.Errorf("duplicate enum %q", tok)
+		}
```
Comment (lblackstone) [inline provider/pkg/gen/schema.go:743]: You could move this check into the for loop to return early on failure.

**Review by lblackstone — APPROVED** [review]
🎉

File: examples/pubsub-ts/index.ts (hunk 1 above)
Comment (mikhailshilkov) [reply examples/pubsub-ts/index.ts:51]: It is an enum, there are two ways to express them in TS: a string or a constant. I changed the other example to a const and this example is a string (but `"avro"` wouldn't compile).

**Pull request merged**